pallets/gear-scheduler/src/migrations/v3_remove_program_pause_tasks.rs

```diff
@@ -0,0 +1,251 @@
+// This file is part of Gear.
+
+// Copyright (C) 2023-2024 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{Config, Pallet};
+use frame_support::{
+    traits::{Get, GetStorageVersion, OnRuntimeUpgrade, StorageVersion},
+    weights::Weight,
+};
+use sp_std::marker::PhantomData;
+
+#[cfg(feature = "try-runtime")]
+use {
+    crate::pallet::TaskPool,
+    frame_support::ensure,
+    sp_runtime::{
+        codec::{Decode, Encode},
+        TryRuntimeError,
+    },
+    sp_std::vec::Vec,
+};
+
+const MIGRATE_FROM_VERSION: u16 = 2;
+const MIGRATE_TO_VERSION: u16 = 3;
+const ALLOWED_CURRENT_STORAGE_VERSION: u16 = 3;
+
+pub struct MigrateRemoveProgramPauseTasks<T: Config>(PhantomData<T>);
```

Comment:
actually if we still iterate all over scheduler storage, it may make sense to migrate other idxs so we won't need to support legacy codec indexes for kept task variants